Add Synapse 1.76 blog post (#1635)
matrix-org/matrix.org: gatsby/content/blog/2023/01/2023-01-31-synapse-1.76.mdx

---
date: '2023-01-31'
title: Synapse 1.76 released
categories:
- Releases
author: H. Shay
---

Greetings matrix fans! We've published Synapse [version 1.76](https://github.com/matrix-org/synapse/releases/tag/v1.76.0)
as the new stable release this week. Synapse admins are encouraged to upgrade

> to it at their convenience. Please take a look at the [upgrade notes](https://github.com/matrix-org/synapse/blob/release-v1.76/docs/upgrade.md#faster-joins-are-enabled-by-default)

Reviewer: We usually point at the doc website for the upgrade notes I believe, https://matrix-org.github.io/synapse/v1.76/upgrade.html#upgrading-to-v1760

for any important information about upgrading.

## Announcements

The big news is that faster joins for all has arrived! When joining a room for the first time,
Synapse 1.76.0 will request a partial join from the other server by default. Previously, server admins
had to opt-in to this using an experimental config flag.

Server admins can opt out of this feature for the time being by setting

```
experimental:
    faster_joins: false
```

in their server config.

For everyone else, after a faster join Synapse considers that room "partially joined". In this state, you should be able to

- read incoming messages;
- see incoming state changes, e.g. room topic changes; and
- send messages, if the room is unencrypted.

Synapse has to spend more effort to complete the join in the background. Once this finishes, you will be able to

> - send messages, if the room is in encrypted;

Reviewer: is the `in` unwanted or it is an English idiom I don't know about ?

Room version 10 is now the default room version in Synapse 1.76.0. This room version builds on v9 to introduce a new
`knock_restricted` join rule, allowing prospective members to more easily join such a room. The `knock_restricted` join
rule allows for prospective members of a room to join either through knocking or through "join_restrictions", which allows
prospective members to join a restricted room based on whether they are a joined member of another room. You can read more
about restricted rooms [here](https://spec.matrix.org/v1.5/client-server-api/#restricted-rooms).

### Synapse now implements experimental support for [MSC3930](https://github.com/matrix-org/matrix-spec-proposals/pull/3930) Push rules for Polls

Synapse has added some new, default push rules for [MSC3881](https://github.com/matrix-org/matrix-spec-proposals/pull/3381)-esque event types.

This allows for notifications from polls, ie when a poll starts or ends.

## Everything else

See the full [changelog](https://github.com/matrix-org/synapse/releases/tag/v1.76.0), for a
complete list of changes in the release.

Synapse is a Free and Open Source Software project, and we'd like to extend our

> thanks to everyone who contributed to these releases, including: FSG-Cat and thezaidbintariq.

Reviewer: I found the script we were using to fetch all external contributors for a release, it's probably a good idea to use it here.
https://github.com/babolivier/external-github-contributors

---
date: '2023-01-31'
title: Synapse 1.76 released
categories:
- Releases
author: H. Shay
---

Greetings matrix fans! We've published Synapse [version 1.76](https://github.com/matrix-org/synapse/releases/tag/v1.76.0)
as the new stable release this week. Synapse admins are encouraged to upgrade
to it at their convenience. Please take a look at the [upgrade notes](https://matrix-org.github.io/synapse/v1.76/upgrade.html#upgrading-to-v1760)
for any important information about upgrading.

## Announcements

The big news is that faster joins for all has arrived! When joining a room for the first time,
Synapse 1.76.0 will request a partial join from the other server by default. Previously, server admins
had to opt-in to this using an experimental config flag.

Server admins can opt out of this feature for the time being by setting

```
experimental:
    faster_joins: false
```

in their server config.

For everyone else, after a faster join Synapse considers that room "partially joined". In this state, you should be able to

> - read incoming messages;

Reviewer 2: do/should we note that this only applies to clients which use lazyloaded members?

Author: I don't have a strong opinion on this, what do you think we should say here?

Reviewer 2: I guess *most people* will be using Element, but on Matrix.org I like to think we try to be quite neutral so I think it would be worth pointing out the caveat. I'd probably write as 'read incoming messages (on clients which support lazy-loaded members in `/sync`)'

Author: Done, thanks for the suggestion and review!

- see incoming state changes, e.g. room topic changes; and
- send messages, if the room is unencrypted.

Synapse has to spend more effort to complete the join in the background. Once this finishes, you will be able to

- send messages, if the room is encrypted;
- retrieve room history from before your join, if permitted by the room settings; and
- access the full list of room members.

Practically, this means that the experience of joining a large room over federation should be greatly improved. In past
versions of Synapse, joining a large room such as Matrix HQ over federation could take up to 30 mins. This time has been

> cut to ~20-30 seconds in the new release. Those of you who have been watching this know this has been a huge undertaking,

Reviewer 2: This kind of sets the wrong expectation a bit, I'd try to be more careful so that people know that the room is only *partially* joined after 20-30 seconds, but it's not quite the real deal.
(I've seen people in community rooms confused about this — 'faster joins are a bit slow' etc etc)

Author: I've changed it to reflect partially joining.

> big shout out to the team for getting it over the line! We're excited the community to try it, as always if you encounter

Reviewer 2: should we really shout out to ourselves? :p

Reviewer 2: hahah, fine, not how I would take it but I'm not writing the blog post ;-) :P

any problems you are encouraged to file an [issue](https://github.com/matrix-org/synapse/issues/new?assignees=&labels=&template=BUG_REPORT.yml).

### Changes to the account data replication streams

Please note that Synapse has changed the format of the account data and devices replication streams (between workers).
This is a forwards- and backwards-incompatible change: v1.75 workers cannot process account data replicated by v1.76
workers, and vice versa.

Once all workers are upgraded to v1.76 (or downgraded to v1.75), account data and device replication will resume as normal.

## The new stuff

### The default room version has now been updated to [v10](https://spec.matrix.org/v1.5/rooms/v10/)

Room version 10 is now the default room version in Synapse 1.76.0. This room version builds on v9 to introduce a new
`knock_restricted` join rule, allowing prospective members to more easily join such a room. The `knock_restricted` join

> rule allows for prospective members of a room to join either through knocking or through "join_restrictions", which allows

Reviewer 2: ```suggestion
rule allows for prospective members of a room to join either through knocking or through `join_restrictions`, which allows
```
either that or lose the underscore — basically either human speak or computer speak, but choose one and use the right style

---
date: '2023-01-31'
title: Synapse 1.76 released
categories:
- Releases
author: H. Shay
---

Greetings matrix fans! We've published Synapse [version 1.76](https://github.com/matrix-org/synapse/releases/tag/v1.76.0)
as the new stable release this week. Synapse admins are encouraged to upgrade
to it at their convenience. Please take a look at the [upgrade notes](https://github.com/matrix-org/synapse/blob/release-v1.76/docs/upgrade.md#faster-joins-are-enabled-by-default)
for any important information about upgrading.

## Announcements

The big news is that faster joins for all has arrived! When joining a room for the first time,
Synapse 1.76.0 will request a partial join from the other server by default. Previously, server admins
had to opt-in to this using an experimental config flag.

Server admins can opt out of this feature for the time being by setting

experimental:

>     faster_joins: false

Reviewer 3: This should probably be a YAML code block. I think it is just markdown?

```yaml
experimental:
    faster_joins: false 
```

in their server config.

For everyone else, after a faster join Synapse considers that room "partially joined". In this state, you should be able to

>     - read incoming messages;
>     - see incoming state changes, e.g. room topic changes; and
>     - send messages, if the room is unencrypted.

Reviewer 3: ```suggestion
- read incoming messages;
- see incoming state changes, e.g. room topic changes; and
- send messages, if the room is unencrypted.
```

Synapse has to spend more effort to complete the join in the background. Once this finishes, you will be able to

>     - send messages, if the room is in encrypted;
>     - retrieve room history from before your join, if permitted by the room settings; and
>     - access the full list of room members.

Reviewer 3: ```suggestion
- send messages, if the room is in encrypted;
- retrieve room history from before your join, if permitted by the room settings; and
- access the full list of room members.
```

Practically, this means that the experience of joining a large room over federation should be greatly improved. In past

> versions of Synapse, joining a large room such as MatrixHQ over federation could take up to 30 mins. This time has been

Reviewer 3: ```suggestion
versions of Synapse, joining a large room such as Matrix HQ over federation could take up to 30 mins. This time has been
```

cut to ~20-30 seconds in the new release. Those of you who have been watching this know this has been a huge undertaking,
big shout out to the team for getting it over the line! We're excited the community to try it, as always if you encounter

> any problems you are encouraged to file an [issue](https://github.com/matrix-org/synapse/issues).

Reviewer 3: Maybe go right to the new issue form?

```suggestion
any problems you are encouraged to file an [issue](https://github.com/matrix-org/synapse/issues/new?assignees=&labels=&template=BUG_REPORT.yml).
```

### Changes to the account data replication streams

Please note that Synapse has changed the format of the account data and devices replication streams (between workers).
This is a forwards- and backwards-incompatible change: v1.75 workers cannot process account data replicated by v1.76
workers, and vice versa.

Once all workers are upgraded to v1.76 (or downgraded to v1.75), account data and device replication will resume as normal.

## The new stuff

> ### The default room version has now been updated to v10

Reviewer 3: ```suggestion
### The default room version has now been updated to [v10](https://spec.matrix.org/v1.5/rooms/v10/)
```

Room version 10 is now the default room version in Synapse 1.76.0. This room version builds on v9 to introduce a new
`knock_restricted` join rule, allowing prospective members to more easily join such a room. The `knock_restricted` join
rule allows for prospective members of a room to join either through knocking or through "join_restrictions", which allows
prospective members to join a restricted room based on whether they are a joined member of another room. You can read more
about restricted rooms [here](https://spec.matrix.org/v1.5/client-server-api/#restricted-rooms).

### Synapse now implements experimental support for [MSC3930](https://github.com/matrix-org/matrix-spec-proposals/pull/3930) Push rules for Polls

Synapse has added some new, default push rules for [MSC3881](https://github.com/matrix-org/matrix-spec-proposals/pull/3381)-esque event types.
This allows for notifications from polls, ie when a poll starts or ends.

> You can enable the functionality by adding the following to your config in the experimental section:

Reviewer 3: I suspect we don't want to encourage enabling these?

Author: Ah good to know.
